/*This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:
Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear
*/
From 4994a2299f407669ef6576e30edaf782c8aba064 Mon Sep 17 00:00:00 2001
From: pvijay <pvijay@qti.qualcomm.com>
Date: Wed, 16 Jul 2025 17:40:33 +0530
Subject: [PATCH] DRM : Fix the negative array size for 32 bit

fix the negative array size for the 32 bit
compilation
while compilation for 64 bit it assert for
64 bit architecture so it failed for 32 bit

Signed-off-by: pvijay <pvijay@qti.qualcomm.com>
Upstream-Status: Inappropriate [Downstream]
---
 libdrm_macros.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/libdrm_macros.h b/libdrm_macros.h
index 0dca8273..e3cc948c 100644
--- a/libdrm_macros.h
+++ b/libdrm_macros.h
@@ -73,6 +73,7 @@ static inline void *drm_mmap(void *addr, size_t length, int prot, int flags,
 
 static inline int drm_munmap(void *addr, size_t length)
 {
+#if defined (__aarch_64__)
    /* Copied from configure code generated by AC_SYS_LARGEFILE */
 #define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + \
                      (((off_t) 1 << 31) << 31))
@@ -80,6 +81,15 @@ static inline int drm_munmap(void *addr, size_t length)
                  LARGE_OFF_T % 2147483647 == 1);
 #undef LARGE_OFF_T
 
+#else
+#define LARGE_OFF_T ((((off_t) 1 << 15) << 15) - 1 + \
+                     (((off_t) 1 << 15) << 15))
+   STATIC_ASSERT(LARGE_OFF_T % 32767 == 1 &&
+                 LARGE_OFF_T % 32767 == 1);
+#undef LARGE_OFF_T
+
+#endif
+
    return munmap(addr, length);
 }
 #endif
-- 
2.34.1

